src/Commands/ExportCommand.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace DouCollector\Commands;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use DouCollector\DOU;
+use Spatie\ArrayToXml\ArrayToXml;
+
+class ExportCommand extends Command
+{
+    const FORMAT_WHITELIST = ["json", "xml"];
+
+    protected function configure(): void
+    {
+        $this
+            ->setName('export')
+            ->setDescription('Exporta dados do DOU')
+            ->addOption(
+                'date',
+                'd',
+                InputOption::VALUE_REQUIRED,
+                'Data para realizar importação, exemplo: 30-12-2019'
+            )
+            ->addOption(
+                'keywords',
+                'k',
+                InputOption::VALUE_REQUIRED,
+                'Lista de palavras-chave para importar, separadas por vírgula.'
+            )
+            ->addOption(
+                'format',
+                'f',
+                InputOption::VALUE_OPTIONAL,
+                'O formato de saída (json, xml). [default: "json"]'
+            )
+            ->addOption(
+                'maxRequests',
+                'm',
+                InputOption::VALUE_OPTIONAL,
+                'Máximo de requisições que serão feitas, se 0 fará quantas for necessário. [default: 3]'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $date = $input->getOption('date');
+        $keywords = explode(",", $input->getOption('keywords'));
+        $format = $input->getOption('format') ?? 'json';
+        $maxRequests = $input->getOption('maxRequests') ?? 3;
+
+        if (!$date) {
+            $output->writeln('A opção "date" é obrigatória.');
+            return Command::FAILURE;            
+        }
+        
+        if (empty($keywords) || $keywords[0] == "") {
+            $output->writeln('A opção "keywords" é obrigatória.');
+            return Command::FAILURE;            
+        }
+
+        if (!in_array($format, self::FORMAT_WHITELIST)) {
+            $output->writeln([
+                "Formato \"{$format}\" é inválido!",
+                'Os formatos válidos são: ' . implode(", ", self::FORMAT_WHITELIST)
+            ]);
+            return Command::FAILURE;
+        }
+
+        $DOU = new DOU([
+            'maxRequests' => $maxRequests
+        ]);
+        foreach ($DOU->collectData($date, $keywords) as $result) {
+            $results[] = $result;
+        }
+        
+        $json = json_encode($results);
```

Comment:
aqui faz o encode mas se for xml faz decode, este encode deveria estar então no formato json

Reply:
Então, no caso eu faço esse encode para transformar em string e ai poder fazer o decode!
Se tento usar o $results ao invés do $array, ele da um erro de que stdclass não pode ser utilizado.